compose/compose.go

```diff
@@ -648,16 +678,66 @@ func main() {
 		services[svc.name] = svc
 	}
 
+	// Alpha Customization
+	customAlphas := make(map[int]string)
+	if len(opts.CustomAlphaOptions) > 0 {
+		for _, flag := range opts.CustomAlphaOptions {
+			splits := strings.SplitN(flag, ":", 2)
+			if len(splits) != 2 {
+				fatal(errors.Errorf(" --custom_alpha_flags option requires string in index:options format."))
+			}
+			custIdx, err := strconv.Atoi(splits[0])
+			if err != nil {
+				fatal(errors.Errorf(" --custom_alpha_flags captured erros while parsing index value %v", err))
+			}
+			customAlphas[custIdx] = splits[1]
+		}
+	}
+
+	alphaGroups := make(map[int]int)
+	if len(opts.CustomGroups) > 0 {
+		for _, flag := range opts.CustomGroups {
+			splits := strings.SplitN(flag, ":", 2)
+			if len(splits) != 2 {
+				fatal(errors.Errorf(" --custom_groups option requires string in index:groupNum format."))
+			}
+			custIdx, errIndex := strconv.Atoi(splits[0])
+			custValue, errValue := strconv.Atoi(splits[1])
+			if errIndex != nil || errValue != nil {
+				fatal(errors.Errorf(" --custom_groups captured erros while parsing index value %v", err))
+			}
+			alphaGroups[custIdx] = custValue
+		}
+	}
+
+	gid := 0
+	idxNum := 0
 	for i := 1; i <= opts.NumAlphas; i++ {
-		gid := int(math.Ceil(float64(i) / float64(opts.NumReplicas)))
-		rs := fmt.Sprintf("idx=%d; group=%d", i, gid)
-		svc := getAlpha(i, rs)
+		if val, ok := alphaGroups[i]; ok {
```

Comment:
Why do you need this logic? Say if we want 2 groups each with 1 alpha. Then we can set the replicas=1, alphas=2, right?